controllers/operator/openstack_controller.go

```diff
@@ -340,6 +353,37 @@ func (r *OpenStackReconciler) countDeployments(ctx context.Context, instance *op
 	return count, nil
 }
 
+// checkServiceEndpoints -
+func (r *OpenStackReconciler) checkServiceEndpoints(ctx context.Context, instance *operatorv1beta1.OpenStack) (ctrl.Result, error) {
+
+	// NOTE: this is a static list for all operators with webhooks enabled
+	endpointNames := []string{"openstack-operator-webhook-service", "infra-operator-webhook-service", "openstack-baremetal-operator-webhook-service"}
+
+	for _, endpointName := range endpointNames {
+		endpoints := &corev1.Endpoints{}
+		err := r.Client.Get(ctx, client.ObjectKey{Name: endpointName, Namespace: instance.Namespace}, endpoints)
```

Comment:
Do we maybe need to add kubebuilder annotations to grant us cluster-scoped permission to `get` or `list` `Endpoint`s?

Reply:
ack, yes we are missing permissions